src/com/google/javascript/jscomp/Es6SyntacticScopeCreator.java

```diff
@@ -100,13 +100,27 @@ private void scanRoot(Node n) {
       }
 
       // Since we create a separate scope for body, stop scanning here
+    } else if (n.isClass()) {
+      if (scope.getParent() != null) {
+        inputId = NodeUtil.getInputId(n);
+      }
+
+      final Node classNameNode = n.getFirstChild();
+      // Bleed the class name into the scope, if it hasn't
+      // been declared in the outer scope.
+      if (!classNameNode.isEmpty()) {
+        String className = classNameNode.getString();
+        if (!className.isEmpty() && NodeUtil.isClassExpression(n)) {
```

Comment:
When class has no name, it has an EMPTY node rather than an empty string like function nodes.